build: new user boost wip

matthew-garrett opened the pull request:
build: adds products table

build: new rocket img

build: adds boost values to tx and card pages

jeffywu commented on apps/web/src/containers/NewUserView/index.tsx:
```diff
@@ -0,0 +1,365 @@
+import newcomer from './newcomer.png';
+
+export const NewUserView = () => {
+  const landingTheme = useNotionalTheme(THEME_VARIANTS.LIGHT, 'landing');
```
Should we redirect them to error if they are not on the whitelist?

jeffywu commented on packages/common/icons/src/lib/rocket-icon/rocket-icon.tsx:
```diff
@@ -0,0 +1,35 @@
+    <SvgIcon viewBox="0 0 20 20" {...props}>
+      <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 146 146">
+        <path
+          fill-rule="evenodd"
```
Can you fix all the fill-rule and clip-rule errors that show up in the console?

jeffywu commented on packages/features/portfolio/src/containers/portfolio-holdings/use-detailed-holdings.tsx:
```diff
@@ -106,9 +117,10 @@ export function useDetailedHoldingsTable() {
-          // const pointsPerDay = getPointsPerDay(b);
-          // const totalPoints =
-          //   arbPoints?.find(({ token }) => token === b.tokenId)?.points || 0;
+          const isBoostedToken = checkBoostedToken(
```
Can we call this `isStarterBoost` instead and `checkStarterBoostToken` or something like that

jeffywu commented on packages/notionable/src/global/account/communities.ts:
```diff
@@ -97,3 +97,7 @@ export function checkSanctionedAddress(account: string) {
 }
+
+export function checkNewUserAddress(account: string) {
+  return newUserAddresses.includes(account);
```
Make sure that all the account address is lower case here and all the addresses in the json are lower case as well.

jeffywu commented on packages/notionable/src/global/global-state.ts:
```diff
@@ -117,6 +117,8 @@ interface AddressState {
   /** Checks if the address is sanctioned on wallet connection */
   isSanctionedAddress: boolean;
+  /** Checks if the address is a new user on wallet connection */
+  isNewUser: boolean;
```
Can we change this to `isStarterBoostUser` instead?

jeffywu commented on packages/util/src/constants.ts:
```diff
@@ -48,6 +49,10 @@ export enum Network {
   arbitrum = 'arbitrum',
 }
 
+export const boostEndDate = new Date('2024-12-16');
```
I would only export one of these two variables depending on which one you are using.

matthew-garrett committed:
fix: remove dd forward (#1237)

fix: force note reinvestment on Sunday (#1239)

build: adds debank lender data (#1238)

* build: adds debank lender data

* build: adds lending value to tracking

* build: addresses comment
fix: add pendle vault addresses (#1240)

Bugfix/analytics links (#1245)

* build: updates links removes broken notional dash

* build: removes log

* build: switches DD mask to allow
fix: some pendle bugs (#1244) (#1248)

Feat/update vault liquidation prices (#1185)

* fix: change liquidation price algorithm

* fix: liquidation price hooks

* fix: remove vault liquidation chart

* fix: vault risk table refactor

* fix: update vault risk table

* fix: update vault risk ui

* fix: update vault charts

* fix: update liquidation prices

* fix: add vault points type

* fix: type errors
Feat/total apy vault updates (#1206)

* build: moves reinvest pill to mui

* build: adds points value tooltips refactors reinvest pill

* fix: add reward apy

---------

Co-authored-by: Jeff Wu <jeff@notional.finance>
Bugfix/pendle bug fixes (#1224)

* fix: manage vault prime borrow rate

* fix: remove header

* fix: post expiry rate

* fix: account max withdraw calculation

* fix: performance chart borrow rate

* fix: adjust leverage errors

* fix: increase vault position risk factor limit

* fix: adding logging for vault liquidation prices
Bugfix/more pendle liquidation price fixes (#1227)

* fix: borrow fee amount

* fix: vault liquidation prices
fix: inverted threshold (#1234)

Feat/add pendle vaults (#1241)

* fix: dd forward headers (#1236)

* fix: add pendle vault addresses

* feat: adding pendle addresess
fix: default pool names (#1242)

build: new user boost wip

build: adds products table

build: new rocket img

build: adds boost values to tx and card pages

build: clean up

build: addresses comments

build: removes my address

jeffywu commented on packages/util/src/constants.ts:
```diff
@@ -261,3 +266,3274 @@ export const WETHAddress: Record<Network, Lowercase<TokenAddress>> = {
   [Network.arbitrum]: toLower(tokens.arbitrum.WETH),
 };
+
+export const NEW_USER_ADDRESSES = [
```
I think we should move this into a file that we fetch via HTTP instead of committing to the repo

jeffywu replied:
@matthew-garrett can you fetch the list from here:
https://registry.notional.finance/newcomer-boost.json

jeffywu committed:
fix: remove addresses

matthew-garrett committed:
build: adds new user list fetch
